src/org/labkey/test/components/glassLibrary/grids/ResponsiveGrid.java

```diff
@@ -291,71 +251,58 @@ public boolean hasSelectColumn()
+     */
+    protected Integer getColumnIndex(String columnHeader)
     {
-        List<String> columnTexts = getColumnNames();
-        int offset = hasSelectColumn() ? 1 : 0;
-        for (int i=0; i< columnTexts.size(); i++ )
+        if (elementCache().indexes == null)
         {
-            if (columnTexts.get(i).equalsIgnoreCase(columnHeader))
-                return i + offset;  // the presence of a select column in the grid will shift everything right by 1
+            getColumnNames();
         }
-        return -1;
+
+        if (elementCache().indexes.containsKey(columnHeader))
+            return elementCache().indexes.get(columnHeader).get("rawIndex");
+        else
+            return -1;
     }
 
     public List<String> getColumnNames()
     {
-        return elementCache().getColumnNames();
+        elementCache().initColumnsAndIndices();
+        return elementCache().columnNames;
     }
 
+    /**
+     * there are ways to get rowMaps without exposing indexes to outside use
+     * @param rowIndex
+     * @return
+     */
+    @Deprecated
     public List<String> getRowValues(int rowIndex)
     {
         // preserves the ordering of the values as they appear in the row.
         if (!hasData())
             throw new IllegalStateException("Attempting to get a row by index, but no rows exist");
 
-        List<String> rowValues = getWrapper().getTexts(Locator.css("td").findElements(getRow(rowIndex)));
-        if (hasSelectColumn())
-        {
-            rowValues.remove(0);
-        }
-        return rowValues;
+        return getRow(rowIndex).getValues();
     }
 
     public Map<String, String> getRowMap(int rowIndex)
     {
-        HashMap<String, String> map = new HashMap<>();
-        List<String> columns = getColumnNames();
-        List<String> rowCellTexts = getRowValues(rowIndex);
-        for (int i=0; i < columns.size(); i++)
-        {
-            map.put(columns.get(i), rowCellTexts.get(i));
-        }
-        return map;
+        return getRow(rowIndex).getRowMap();
     }
 
     public Map<String, String> getRowMap(WebElement row)
     {
-        List<String> tableColumns = getColumnNames();
-        List<String> rowValues = getWrapper().getTexts(Locator.css("td").findElements(row));
-
-        // If it has a select column remove that entry from the rowValues.
-        if(hasSelectColumn())
-        {
-            rowValues.remove(0);
-        }
-
-        Map<String, String> rowMap = new HashMap<>();
-        for (int i=0; i < tableColumns.size(); i++)
-        {
-            rowMap.put(tableColumns.get(i), rowValues.get(i));
-        }
-        return  rowMap;
+        return new GridRow(this, row, getDriver()).getRowMap();
     }
 
     public String getCellValue(int rowIndex, String columnHeader)
     {
-        return getRowMap(rowIndex).get(columnHeader);
+        return getRow(rowIndex).getValue(columnHeader);
     }
 
     public String getCellValue(WebElement row, String columnHeader)
```

Comment:
```suggestion
    public String getCellText(WebElement row, String columnHeader)
```

Reply:
If possible, I'd like to eliminate usage of webElements as input parameters like the one at line 308.  Likewise, on 303 the external reliance on rowIndex.  Because line 308 isn't being used from anywhere, I'm cutting it